content/docs/rfcs/32/README.md

```diff
@@ -0,0 +1,457 @@
+After they are part of the memebrship group, 
+the users can generate ZK-Proofs for their signals and can participate in the application. 
+If a user generates more signals than allowed in an epoch, 
+the user risks being slashed - by revealing his secret and revealing his identity commitment. 
+If financial stake is put in place, the user also risks his stake being taken.
+
+## Registration
+
+Depending on the application requirements, the registration can be implemented in different ways, for example: 
+- centralized registrations, by using a central server
+- decentralized registrations, by using a smart contract
+
+What is important is that the user's identity commitment is stored in a merkle tree, 
+and the users can obtain a merkle proof proving that they are part of the group.
+
+Also depending on the application requirements, 
+usually a financial or social stake is introduced.
+
+An example for financial stake is: For each registration a certain amount of ETH is required.
+An example for social stake is using InterRep as a registry - 
+users need to prove that they have a highly reputable social media account.
+
+The user's identity is composed of:
+
+```
+{
+    identity_secret: secret_component_arr,
+    identity_secret_hash: poseidonHash(identity_secret),
+    identity_commitment: poseidonHash([identity_secret_hash])
+}
+```
+
+For registration, the user needs to submit their `identity_commitment` (along with any additional registration requirements) to the registry party. 
+Upon registration, they should receive `leaf_index` value which represents their position in the merkle tree. 
+Receiving a `leaf_index` is not a hard requirement and is application specific. 
+The other way around is the users calculating the `leaf_index` themselves upon successful registration.
+
+## Signaling
+
+After registration, 
+the users can participate in the application by sending signals to the other participants in a decentralised manner or to a centralised server. 
+Along with their signal, 
+they need to generate a ZK-Proof by using the circuit with the specification described above. 
+
+For generating a proof, 
+the users need to obtain the required parameters or compute them themselves, 
+depending on the application implementation and client libraries supported by the application.
+For example the users can store the memebrship merkle tree on their end and 
+generate a merkle proof whenever they want to generate a signal. 
+
+### Signal hash
+
+The signal hash is generated by hashing the raw signal (or content) using the `keccak256` hash function, 
+converting the hashed value to number and padding it:
+
+```
+  signal: string = "signal"
+  converted: string = toHex(toUtf8Bytes(signal))
+  signal_hash: bigint = BigInt(keccak256(converted)) >> BigInt(8)
+```
+
+
+### Epoch
+
+The epoch (or external nullifier) is also obtained by hasing a raw string value using `keccak256`:
```

Comment:
Rewrite for clarity - if I understand correctly this could be a new section, something like:

### External nullifier

The external nullifier is hashed using `keccak256` in the same way as for `epoch` above.